stdlib/net-http/0/net-http.rbs

```diff
@@ -0,0 +1,1868 @@
+
+    RESPONSE_HAS_BODY: bool
+  end
+
+  # See Net::HTTPGenericRequest for attributes and methods.
+  class HTTP::Move < HTTPRequest
+    METHOD: String
+
+    REQUEST_HAS_BODY: bool
+
+    RESPONSE_HAS_BODY: bool
+  end
+
+  # See Net::HTTPGenericRequest for attributes and methods.
+  class HTTP::Lock < HTTPRequest
+    METHOD: String
+
+    REQUEST_HAS_BODY: bool
+
+    RESPONSE_HAS_BODY: bool
+  end
+
+  # See Net::HTTPGenericRequest for attributes and methods.
+  class HTTP::Unlock < HTTPRequest
+    METHOD: String
+
+    REQUEST_HAS_BODY: bool
+
+    RESPONSE_HAS_BODY: bool
+  end
+
+  class HTTPResponse
+    # true if the response has a body.
+    def self.body_permitted?: () -> bool
+
+    public
+
+    include Net::HTTPHeader
+
+    # The HTTP version supported by the server.
+    attr_reader http_version: String
+
+    # The HTTP result code string. For example, '302'.  You can also
+    # determine the response type by examining which response subclass
+    # the response object is an instance of.
+    attr_reader code: String
+
+    # The HTTP result message sent by the server. For example, 'Not Found'.
+    attr_reader message: String
+
+    alias msg message
+
+    # The URI used to fetch this response.  The response URI is only available
+    # if a URI was used to create the request.
+    attr_reader uri: (URI::Generic | nil)
+
+    # Set to true automatically when the request did not contain an
+    # Accept-Encoding header from the user.
+    attr_accessor decode_content: bool
+
+    def inspect: () -> String
+
+    def code_type: () -> untyped
+
+    def error!: () -> untyped
+
+    def error_type: () -> (Net::HTTPError | Net::HTTPServerException | Net::HTTPRetriableError | Net::HTTPFatalError)
+
+    # Raises an HTTP error if the response is not 2xx (success).
+    def value: () -> (nil | untyped)
+
+    def uri=: (URI::Generic uri) -> void
+
+    # Gets the entity body returned by the remote HTTP server.
+    #
+    # If a block is given, the body is passed to the block, and
+    # the body is provided in fragments, as it is read in from the socket.
+    #
+    # If +dest+ argument is given, response is read into that variable,
+    # with <code>dest#<<</code> method (it could be String or IO, or any
+    # other object responding to <code><<</code>).
+    #
+    # Calling this method a second or subsequent time for the same
+    # HTTPResponse object will return the value already read.
+    #
+    #   http.request_get('/index.html') {|res|
+    #     puts res.read_body
+    #   }
+    #
+    #   http.request_get('/index.html') {|res|
+    #     p res.read_body.object_id   # 538149362
+    #     p res.read_body.object_id   # 538149362
+    #   }
+    #
+    #   # using iterator
+    #   http.request_get('/index.html') {|res|
+    #     res.read_body do |segment|
+    #       print segment
+    #     end
+    #   }
+    #
+    def read_body: (?untyped? dest) { (String parameter) -> untyped } -> String
```

Comment:
The doc says `<<` is used for writing the result to `dest`. So, we can write it in the type by interface.

```rbs
interface _Dest
  def <<: (String) -> void
end

......

def read_body: () -> String
             | (_Dest dest) -> String
             | () { (String) -> void } -> String
```